caas/kubernetes/provider/builtin.go

```diff
@@ -38,43 +38,41 @@ func attemptMicroK8sCloud(cmdRunner CommandRunner) (cloud.Cloud, error) {
 	return k8sCloud, err
 }
 
-func attemptMicroK8sCredential(cmdRunner CommandRunner) (cloud.Credential, error) {
+func attemptMicroK8sCredential(cmdRunner CommandRunner) (jujucloud.Credential, error) {
 	microk8sConfig, err := getLocalMicroK8sConfig(cmdRunner)
 	if err != nil {
-		return cloud.Credential{}, err
+		return jujucloud.Credential{}, err
 	}
 
 	k8sConfig, err := k8scloud.ConfigFromReader(bytes.NewReader(microk8sConfig))
 	if err != nil {
-		return cloud.Credential{}, errors.Annotate(err, "processing microk8s config to make juju credentials")
+		return jujucloud.Credential{}, errors.Annotate(err, "processing microk8s config to make juju credentials")
 	}
 
 	contextName, err := k8scloud.PickContextByClusterName(k8sConfig, caas.MicroK8sClusterName)
 	if err != nil {
-		return cloud.Credential{}, errors.Trace(err)
+		return jujucloud.Credential{}, errors.Trace(err)
 	}
 
 	context := k8sConfig.Contexts[contextName]
 	resolver := clientconfig.GetJujuAdminServiceAccountResolver(jujuclock.WallClock)
 	conf, err := resolver(caas.K8sCloudMicrok8s, k8sConfig, contextName)
 	if err != nil {
-		return cloud.Credential{}, errors.Annotate(err, "resolving microk8s credentials")
+		return jujucloud.Credential{}, errors.Annotate(err, "resolving microk8s credentials")
 	}
 
 	return k8scloud.CredentialFromKubeConfig(context.AuthInfo, conf)
 }
 
 func getLocalMicroK8sConfig(cmdRunner CommandRunner) ([]byte, error) {
-	result, err := cmdRunner.RunCommands(exec.RunParams{
-		Commands: "which microk8s.config",
-	})
-	if err != nil || result.Code != 0 {
+	_, err := osexec.LookPath("microk8s")
```

Comment:
Rather than importing and using `os/exec` and using that directly, it's better to extend CommandRunner interface and, like for `RunCommands`, use `github.com/juju/utils/v2/exec`